buildbot_nix/__init__.py

```diff
@@ -964,24 +974,26 @@ def nix_eval_config(
     )
 
 
-async def do_register_gcroot_if(s: steps.BuildStep) -> bool:
+async def do_register_gcroot_if(
+    s: steps.BuildStep, branch_config: models.BranchConfigDict
+) -> bool:
     gc_root = await util.Interpolate(
         "/nix/var/nix/gcroots/per-user/buildbot-worker/%(prop:project)s/%(prop:attr)s"
     ).getRenderingFor(s.getProperties())
-    out_path = await util.Property("out_path").getRenderingFor(s.getProperties())
-    default_branch = await util.Property("default_branch").getRenderingFor(
-        s.getProperties()
-    )
+    out_path = await s.getProperty("out_path")
+
+    return branch_config.do_register_gcroot(
+        await s.getProperty("default_branch"), await s.getProperty("branch")
+    ) and await s.getProperty("event") == "push" \
```

Comment:
regarding https://github.com/nix-community/buildbot-nix/pull/312#issuecomment-2405883528 i expect this to prevent registering gcroots for PRs unless the source branch of the PR is one of the branches that match the regex and has set `register_gcroots = true`.

Reply:
We cannot match on source branches sadly

Reply:
Without making a network requests to girhubs api, or maybe, hm ill check it again, there might be a way. Maybe it arrives over the webhook but buildbot discards it?

Reply:
i meant that if this branch that receives the push happens be a source branch for a PR then it register a GC root. under all other circumstances i foresee this to not register GC rootfs for PRs.

Reply:
The pr branch names are `pull/number` or smth like that

Reply:
got it, then i think your concern from the linked comment doesn't apply because the branch name won't ever match in case of a PR.